db/db_impl_write.cc

```diff
@@ -607,27 +611,6 @@ void DBImpl::WriteCallbackStatusCheck(const Status& status) {
   }
 }
 
-void DBImpl::MemTableInsertStatusCheck(const Status& status) {
-  // A non-OK status here indicates that the state implied by the
-  // WAL has diverged from the in-memory state.  This could be
-  // because of a corrupt write_batch (very bad), or because the
-  // client specified an invalid column family and didn't specify
-  // ignore_missing_column_families.
-  if (!status.ok()) {
```

Comment:
According to description for `paranoid_checks` we should only stop writes if `paranoid_checks == true`.
https://github.com/facebook/rocksdb/blob/master/include/rocksdb/options.h#L338-L344

The memtable status check to set bg_error was added in this commit, but it didn't check if `paranoid_checks` is set.
6ce42dd0750def33aaa632d064ba0a097ac05b14

It seems to me for the behavior to be consistent we should check if `paranoid_checks` is set for memtable error as well. It is okay to me to keep the existing behavior, but it would make the logic a bit weird. Opinions?